sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/measure/MeasureValueCoder.java

```diff
@@ -38,57 +36,22 @@ public MeasureValueCoder(MetricFinder metricFinder) {
 
   @Override
   public void put(Value value, Object object, CoderContext context) {
-    Measure<?> m = (Measure) object;
-    value.putUTF(m.getMetricKey());
-    value.put(m.getValue());
-    putUTFOrNull(value, m.getData());
-    putUTFOrNull(value, m.getDescription());
-    value.putString(m.getAlertStatus() != null ? m.getAlertStatus().name() : null);
-    putUTFOrNull(value, m.getAlertText());
-    value.putDate(m.getDate());
-    value.put(m.getVariation1());
-    value.put(m.getVariation2());
-    value.put(m.getVariation3());
-    value.put(m.getVariation4());
-    value.put(m.getVariation5());
-    putUTFOrNull(value, m.getUrl());
-    Integer personId = m.getPersonId();
-    value.put(personId != null ? personId.intValue() : null);
-    PersistenceMode persistenceMode = m.getPersistenceMode();
-    value.putString(persistenceMode != null ? persistenceMode.name() : null);
-  }
-
-  private static void putUTFOrNull(Value value, @Nullable String utfOrNull) {
-    if (utfOrNull != null) {
-      value.putUTF(utfOrNull);
-    } else {
-      value.putNull();
-    }
+    DefaultMeasure<?> m = (DefaultMeasure<?>) object;
+    org.sonar.api.batch.measure.Metric<?> metric = m.metric();
+    value.putString(metric.key());
+    value.put(m.value());
   }
 
   @Override
   public Object get(Value value, Class clazz, CoderContext context) {
-    Measure<?> m = new Measure();
     String metricKey = value.getString();
-    org.sonar.api.batch.measure.Metric metric = metricFinder.findByKey(metricKey);
+    org.sonar.api.batch.measure.Metric<?> metric = metricFinder.findByKey(metricKey);
     if (metric == null) {
       throw new IllegalStateException("Unknow metric with key " + metricKey);
     }
-    m.setMetric((org.sonar.api.measures.Metric) metric);
-    m.setRawValue(value.isNull(true) ? null : value.getDouble());
-    m.setData(value.getString());
-    m.setDescription(value.getString());
-    m.setAlertStatus(value.isNull(true) ? null : Metric.Level.valueOf(value.getString()));
-    m.setAlertText(value.getString());
-    m.setDate(value.getDate());
-    m.setVariation1(value.isNull(true) ? null : value.getDouble());
-    m.setVariation2(value.isNull(true) ? null : value.getDouble());
-    m.setVariation3(value.isNull(true) ? null : value.getDouble());
-    m.setVariation4(value.isNull(true) ? null : value.getDouble());
-    m.setVariation5(value.isNull(true) ? null : value.getDouble());
-    m.setUrl(value.getString());
-    m.setPersonId(value.isNull(true) ? null : value.getInt());
-    m.setPersistenceMode(value.isNull(true) ? null : PersistenceMode.valueOf(value.getString()));
+    DefaultMeasure<?> m = new DefaultMeasure()
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "m". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1488)